client/src/app/services/profile.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ProfileService {
+
+  constructor() { }
+
+  publications: Array<any> = [];
+
+  getPublications() {
+    this.publications = [{
```

Comment:
don't bother setting "this.publications" every time you call the getPublications method, just set it when initializing the variable and return the same variable every time